src/containerapp/azext_containerapp/_compose_utils.py

```diff
@@ -5,10 +5,386 @@
+            memory = str(resources.reservations.memory.as_gigabytes())
+    elif service.mem_reservation is not None:
+        memory = str(service.mem_reservation.as_gigabytes())
+    
+    # Default to 2Gi if not specified
+    if memory is None:
+        memory = '2'
+    return memory
+
+
+def resolve_port_or_expose_list(ports, name):
+    if len(ports) > 1:
+        message = f"You have more than one {name} mapping defined in your docker-compose file."
+        message += " Which port would you like to use? "
+        choice_index = prompt_choice_list(message, ports)
+
+    return ports[choice_index]
+
+
+def resolve_ingress_and_target_port(service):
+    # External Ingress Check
+    if service.ports is not None:
+        ingress_type = "external"
+
+        if len(service.ports) == 1:
+            target_port = service.ports[0].target
+        else:
+            ports_list = []
+
+            for p in service.ports:
+                ports_list.append(p.target)
+            target_port = resolve_port_or_expose_list(ports_list, "port")
+
+    # Internal Ingress Check
+    elif service.expose is not None:
+        ingress_type = "internal"
+
+        if len(service.expose) == 1:
+            target_port = service.expose[0]
+        else:
+            target_port = resolve_port_or_expose_list(service.expose, "expose")
+    else:
+        ingress_type = None
+        target_port = None
+    return (ingress_type, target_port)
+
+
+# Ported from azure-cli commit 2f7ef21a0d6c4afb9f066c0d65affcc84a8b36a4.
+def resolve_service_startup_command(service):
+    startup_command_array = []
+    startup_args_array = []
+    if service.entrypoint is not None:
+        startup_command = service.entrypoint.command_string()
+        startup_command_array.append(startup_command)
+        if service.command is not None:
+            # Preserve individual command items as separate args
+            if hasattr(service.command, '__iter__') and not isinstance(service.command, str):
+                startup_args_array = list(service.command)
+            else:
+                startup_args = service.command.command_string()
+                startup_args_array.append(startup_args)
+    elif service.command is not None:
+        # When there's no entrypoint, command should be args
+        # Preserve individual command items as separate args
+        if hasattr(service.command, '__iter__') and not isinstance(service.command, str):
+            startup_command_array = None
+            startup_args_array = list(service.command)
+        else:
+            startup_args = service.command.command_string()
+            startup_command_array = None
+            startup_args_array.append(startup_args)
+    else:
+        startup_command_array = None
+        startup_args_array = None
+    return (startup_command_array, startup_args_array)
+
+
+# ============================================================================
```

Comment:
If you only copy them from azure-cli, you can import them from azure-cli,for example:
```
from azure.cli.command_modules.containerapp._compose_utils import resolve_service_startup_command, resolve_ingress_and_target_port,
```